test/e2e/apps/statefulset.go

```diff
@@ -756,6 +756,10 @@ var _ = SIGDescribe("StatefulSet", func() {
 			}
 			pod, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(context.TODO(), pod, metav1.CreateOptions{})
 			framework.ExpectNoError(err)
+			ginkgo.By("Waiting until pod " + podName + " will start running in namespace " + f.Namespace.Name)
+			if err := e2epod.WaitForPodNameRunningInNamespace(f.ClientSet, podName, f.Namespace.Name); err != nil {
+				framework.Failf("Pod %v did not start running: %v", podName, err)
+			}
```

Comment:
The test is trying to verify that statefulPodName goes into failed at least once, and that the stateful controller sees that and recreates it, and that the trigger is the scheduling of a conflicting port.  If a human introduces a bug in the stateful set that causes it to never start, or a bug in the kubelet that fails pods incorrectly, or another random failure happens, it might not be the conflicting port (admission) that causes it to get recreated.  By waiting for the first pod to start successfully, we move those failures to earlier (we are more confident that the conflicting port is the cause).  This move means debugging is easier (if the pod fails to start it's more likely a bug in the test, or a bug in kubelet startup, if the pod starts but then conflict doesn't cause a restart, it's because someone broke admission in the kubelet)